Adyen/Core/Components/Base/BasicComponentConfiguration.swift

```diff
@@ -8,21 +8,33 @@ import Foundation
 
 /// Any component's most basic configuration.
 public protocol AnyBasicComponentConfiguration: Localizable {
-    
-    /// The form component style.
-    var style: FormComponentStyle { get }
+
+    /// The payment information.
+    var payment: Payment? { get }
+}
+
+/// Any presentable payment component basic configuration.
+public protocol AnyPresentableComponentConfiguration: AnyBasicComponentConfiguration {
+
+    associatedtype StyleType
```

Comment:
I don[t this is a good idea, we can't use `AnyPresentableComponentConfiguration` or any of its derivatives (subprotocols ) in a function parameter or return type or var type, then what's the point?

Reply:
> we can't use AnyPresentableComponentConfiguration or any of its derivatives (subprotocols ) in a function parameter or return type or var type

this statement is correct, but I doubt we plan to do any of the things above - hence I don't see it as a counter-argument 🤔

However, I will delete AnyPresentableComponentConfiguration - it does not bring much value.
Main change in this PR - `style: FormComponentStyle` is no longer part of `AnyBasicComponentConfiguration`

Reply:
just try not to use `associatedtype`, thats my point, it doesn't add any value :-)

and for `AnyPresentableComponentConfiguration` why does not add any value?